google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/DependencyVersionWithBomInspection.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.apis;
+
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemDescriptor;
+import com.intellij.lang.annotation.HighlightSeverity;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.impl.source.xml.XmlTextImpl;
+import com.intellij.psi.xml.XmlTag;
+import com.intellij.util.IncorrectOperationException;
+import com.intellij.util.xml.DomFileElement;
+import com.intellij.util.xml.GenericDomValue;
+import com.intellij.util.xml.highlighting.DomElementAnnotationHolder;
+import com.intellij.util.xml.highlighting.DomElementsInspection;
+import java.util.Set;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.idea.maven.dom.model.MavenDomDependency;
+import org.jetbrains.idea.maven.dom.model.MavenDomProjectModel;
+
+/**
+ * An {@link DomElementsInspection} that detects google-cloud-java dependencies that have an
+ * explicit version definition when a BOM is defined.
+ *
+ * <p>Provides a quick-fix to strip out the version tag from the dependency.
+ */
+public class DependencyVersionWithBomInspection
+    extends DomElementsInspection<MavenDomProjectModel> {
```

Comment:
It looks like `DomElementsInspection` is a standard inspection for XML files? I don't see any filtering for a specific file names or schemas, so I guess it does checking based on the DomElement type you pass as a parameter?

Reply:
exactly. its based on the DOM model inspection is parameterized by 

e.g. `MavenDomProjectModel` in this case.

Reply:
If you were curious, see `DomElementsInspection#checkFile`. You can see that it first checks to ensure that the file under inspection is of the same class as the DOM model that is passed in (`myDomClasses`); so it will quickly skip all other files.

Reply:
nice, makes it look easy to add another XML element inspection without low-level work!

Reply:
indeed. and if you look at my commit history here, you'll see that I didn't realize that this was available at first. Initially I had a lower level inspector that was based on XmlElement (with no DOM schema). This is much nicer.